modules/ROOT/pages/apikit-odatav4-expand-tutorial.adoc

```diff
@@ -0,0 +1,163 @@
+   `NavigationProperty` of its entity type.
+** Every `<entity-collection-source/>` should support the `in` operator for
+   the `$filter` system query option.
+. For each source where you want to enable _generic expand_ support:
+.. Open the *Capabilities* tab
+.. Edit inline *Expandable navigation properties*
+.. Add the name of each navigation property you want to be expandable using
+   _generic expand_.
+.. Depending on your usecase check *Ignore errors on expand* if you don't want
+   errors during generic expansion to generate errors on the main flow.
+. Enjoy how your existent flows enable `$expand` support for you!
+
+== How _generic expand_ works
+
+Generic expand works by dispatching additional queries to the application once
+the original flow is completed. Because this is done based on the application
+`EDM` extra annotations might be needed in order to add enough context to the
+module:
+
+. The `EntitySet` where the expansion takes place should have a
+  `NavitationPropertyBinding` for every expandable `NavigationProperty` of its
+  `Entity`. +
+  This used to select which `<entity-collection-source/>` will be executed when
+  expanding that property.
+. Each expandable `NavigationProperty` should have a `ReferentialConstraint`
+  attached. +
+  This is used to create the inner query where the value from the `Property`
+  will be matched against the value of the `ReferencedProperty`.
+
+=== Flow execution
+
+. The main flow is executed and completed. A collection of _entities_ is it's
+  result.
+. The OData Module gathers all the properties that need expansion.
+. For each `NavigationProperty`:
+.. Get the _source property_ from its `ReferentialConstraint`.
+.. Get the _referenced property_ from its `ReferentialConstraint`.
+.. Get the _destination entity set_ from its `NavigationPropertyBinding`.
+.. For each of the _entities_ get the resolved _source property_ put all these
+   values on a _source property list_.
+.. Execute the _virtual_ request by invoking the `<entity-collection-source/>`
+   for _destination entity set_ with
+   `$filter=_referenced property_ in (..._source property list_)`.
+
+=== Error handling and error propagation
+
+Error handling for OData v4 applications will still be local to each flow.
+Wether yo chose to use _generic expansion_ or not. Error handling code for a
+given `EntitySet` will belong to the flow implementations for that entity set
+both for normal requests and for the _virtual_ requests observed during
+expansion processing.
+
+Handling errors during the expansion process is not supported, as the expansion
+process runs after the execution of the flow has finalized. For that two options
+are given:
+
+- *Ignore errors during expansion:* Errors during generic expansion will be
+  ignored. +
+  Expansion of collections and single objects will be resolved to empty lists
+  and `null` values respectively.
+- *Propagate errors during expansion:* Errors during generic expansion will
```

Comment:
I would mention that this is the default config

Reply:
Good point!

Reply:
Good point!